src/oom/oomd-util.c

```diff
@@ -337,31 +512,18 @@ int oomd_kill_by_pgscan_rate(Hashmap *h, const char *prefix, bool dry_run, char
                 if (c->pgscan == 0 && c->current_memory_usage == 0)
                         continue;
 
-                r = oomd_cgroup_kill(c->path, /* recurse= */ true, /* dry_run= */ dry_run);
-                if (r == -ENOMEM)
-                        return r; /* Treat oom as a hard error */
-                if (r < 0) {
-                        RET_GATHER(ret, r);
-                        continue; /* Try to find something else to kill */
-                }
-
-                ret = r;
-                r = strdup_to(ret_selected, c->path);
-                if (r < 0)
-                        return r;
+                *ret_selected = TAKE_PTR(c);
 
-                killed = c;
-                break;
+                return 1;
```

Comment:
what does return 1 mean? Usually, it some kind of special condition which needs explicit documentation. Plus, handling at the caller.